docs/change_resource_hierarchy.md

```diff
@@ -0,0 +1,350 @@
+# Guidance for modifications of the resource hierarchy
+
+This guide explains the instructions to change resource hierarchy during Terraform Foundation Example blueprint deployment.
+
+The current deployment scenario of Terraform Foundation Example blueprint considers a flat resource hierarchy where all folders are at the same level and have one folder for each environment. Here is a detailed explanation of each folder:
+
+| Folder | Description |
+| --- | --- |
+| bootstrap | Contains the seed and CI/CD projects that are used to deploy foundation components. |
+| common | Contains projects with cloud resources used by the organization. |
+| production | Environment folder that contains projects with cloud resources that have been promoted into production. |
+| non-production | Environment folder that contains a replica of the production environment to let you test workloads before you put them into production. |
+| development | Environment folder that is used as a development and sandbox environment. |
+
+This document covers a scenario where you can have two or more levels of folders, with an environment-centric focus: `environments -> ... -> business units`.
+
+| Current Hierarchy | Changed Hierarchy |
+| --- | --- |
+| <pre>example-organization/<br>├── fldr-bootstrap<br>├── fldr-common<br>├── <b>fldr-development *</b><br>├── <b>fldr-non-production *</b><br>└── <b>fldr-production *</b><br></pre> | <pre>example-organization/<br>├── fldr-bootstrap<br>├── fldr-common<br>├── <b>fldr-development *</b><br>│   ├── finance<br>│   └── retail<br>├── <b>fldr-non-production *</b><br>│   ├── finance<br>│   └── retail<br>└── <b>fldr-production *</b><br>    ├── finance<br>    └── retail<br></pre> |
+
+## Code Changes - Build Files
+
+Review the `tf-wrapper.sh`. It is a bash script helper responsible for applying  terraform configurations for Terraform Foundation Example blueprint. The `tf-wrapper.sh` script works based on the current branch (see [Branching strategy](../README.md#branching-strategy)) and searches for a folder in the source code where name matches the current branch name. When it finds a folder it applies the terraform configurations. These changes below will make `tf-wrapper.sh` capable of searching deeper for matching folders and complying with your source code folder hierarchy.
```

Comment:
@daniel-cit @felipecrescencio-cit Since we have identified the changes needed could we make these edits directly on the wrapper and ask users to change max depth as needed?

Reply:
We discussed offline to make the changes as a follow up